### PR TITLE
add /bitrate setting to calls

### PR DESCRIPTION
**Pull request by ghost**
When you are in a call, just run '/bitrate X', where X is a value between 6 and 510 (Opus bitrate range in kbps, per the spec).

### CODE REVIEW

**Comment by JFreegman** [comment]
Could you please run this through astyle using the astyle script provided in the base directory?

File: src/audio_call.c [hunk 1]
```diff
@@ -834,6 +835,60 @@ void cmd_sense(WINDOW *window, ToxWindow *self, Tox *m, int argc, char (*argv)[M
...
+        error_str = "Must be in a call";
+        goto on_error;
+    }
+
+    const uint32_t bitrate = atoi(argv[1]);
+
+    if ( bitrate < 6 || bitrate > 510 ) {
+        error_str = "Bitrate falls outside of Opus spec";
+
+        while (true) {}
```
Comment (JFreegman) [inline src/audio_call.c]: What's this?

File: src/audio_call.c [hunk 2]
```diff
@@ -834,6 +835,60 @@ void cmd_sense(WINDOW *window, ToxWindow *self, Tox *m, int argc, char (*argv)[M
...
+
+        goto on_error;
+    }
+
+    TOXAV_ERR_BIT_RATE_SET error;
+    audio_bit_rate_status_cb(CallControl.av, CallControl.calls[self->num].out_idx, bitrate, -1, &error);
+
+    if (error != TOXAV_ERR_BIT_RATE_SET_OK) {
+        switch (error) {
```
Comment (JFreegman) [inline src/audio_call.c]: This switch needs a default case.

File: src/audio_call.c [hunk 3]
```diff
@@ -834,6 +835,60 @@ void cmd_sense(WINDOW *window, ToxWindow *self, Tox *m, int argc, char (*argv)[M
...
+            case TOXAV_ERR_BIT_RATE_SET_SYNC:
+                error_str = "TOXAV_ERROR_BIT_RATE_SET_SYNC";
```
Comment (JFreegman) [inline src/audio_call.c]: These are user-facing error messages, so they should be more descriptive. Look around other parts of the code to see examples. If you're unsure what a specific error code means you can look them up the toxav API.

File: src/audio_call.c [hunk 4]
```diff
@@ -834,6 +835,60 @@ void cmd_sense(WINDOW *window, ToxWindow *self, Tox *m, int argc, char (*argv)[M
...
+    if ( bitrate < 6 || bitrate > 510 ) {
+        error_str = "Bitrate falls outside of Opus spec";
+
+        while (true) {}
+
+        goto on_error;
+    }
+
+    TOXAV_ERR_BIT_RATE_SET error;
+    audio_bit_rate_status_cb(CallControl.av, CallControl.calls[self->num].out_idx, bitrate, -1, &error);
```
Comment (JFreegman) [inline src/audio_call.c]: The second parameter should be `self->num`. You're passing the friend number, not the input device index.

**Review by JFreegman — CHANGES_REQUESTED** [review]
Two things. 

1: You'll need to add an item in the help menu for the /bitrate command. It should be in the Chat Commands/Audio section.

2: The /bitrate command should be added to the tab-complete command list in chat.c. Make sure to update the appropriate constants above the list as well.

File: src/audio_call.c (hunk 1 above)
Comment (ghost) [reply src/audio_call.c]: whoops, that's some leftover nonsense from debugging. my bad

File: src/audio_call.c (hunk 4 above)
Comment (ghost) [reply src/audio_call.c]: fixed that in a second commit when I tested it, it worked with EchoBot when I called it, so I thought it was correct at the time

File: src/audio_call.c (hunk 2 above)
Comment (ghost) [reply src/audio_call.c]: added

File: src/audio_call.c (hunk 3 above)
Comment (ghost) [reply src/audio_call.c]: modified those as well

File: src/help.c [hunk 5]
```diff
@@ -242,6 +242,7 @@ static void help_draw_chat(ToxWindow *self)
     wprintw(win, "  /sdev <type> <id>          : Change active device\n");
     wprintw(win, "  /mute <type>               : Mute active device if in call\n");
     wprintw(win, "  /sense <n>                 : VAD sensitivity threshold\n");
+    wprintw(win, "  /bitrate <n>               : Set the audio encoding bitrate\n");
```
Comment (JFreegman) [inline src/help.c]: You also need to update the appropriate height parameters in help_onKey(). You'll probably want to do a bit of testing to make sure it looks right.

**Review by JFreegman — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
The latest commit should fix the height issue

**Pull request closed without merging**